# Assignment 6

To measure the psychological state of Americans, a psychology researcher (in 1929) devised
three questions put to 3443 Americans from different walks of life. He asked them whether they
were happy. They could answer only **yes** or **no** to each question. By today’s standards the study
is quite simple, and he did no formal statistical analysis of his data. The frequency data (number
of **yes** and **no** responses within each employment category) from this study are in the file
*Americandata.csv*.

---

You need to use the data and answer the question, that is, **Is the type of profession related to the happiness of American people?**
Report the **χ2** statistic and the *statistical significance of test*, in addition to the contingency table.
Also, explain the results and what they mean. What does the value of **χ2** statistic mean? Is it
good or bad? Is the test statistically significant?

In [7]:
#imports
import pandas as pd
import scipy.stats as stats

#load data
df = pd.read_csv("./Americandata.csv", index_col="Happy")

In [9]:
df.head(20)

,Students,Laborers,Preachers,Physicians,Housewives,Teachers,Lawyers,Musicians
Happy,,,,,,,,
Yes,390,378,35,159,78,108,11,31
No,1610,122,265,51,122,38,64,19


In [10]:
# the way the data is built, is already a contengency table (as shown above when displaying the df)
contingency_table = df

# Run the chi-square test
chi2, p_value, dof, expected = stats.chi2_contingency(contingency_table, correction=False)

In [11]:
# Show expected counts
expected_df = pd.DataFrame(expected, index=contingency_table.index, columns=contingency_table.columns)
print("\nExpected counts:\n", expected_df.round(2))


Expected counts:
        Students  Laborers  Preachers  Physicians  Housewives  Teachers  \
Happy                                                                    
Yes      683.71    170.93     102.56       71.79       68.37     49.91   
No      1316.29    329.07     197.44      138.21      131.63     96.09   

       Lawyers  Musicians  
Happy                      
Yes      25.64      17.09  
No       49.36      32.91  


In [14]:
# Check the assumption
if (expected >= 5).all():
    print("\nAssumption Okay; all counts greater than 5.")
else:
    print("\nWARNING! some expected counts may be less than 5.")

# Report the numbers
print(f"\nChi-square = {chi2:.2f}")
print(f"Degrees of freedom = {dof}")
print(f"p-value = {p_value:.3f}")


Assumption Okay; all counts greater than 5.

Chi-square = 936.14
Degrees of freedom = 7
p-value = 0.000


In [15]:
# Decide using critical value
alpha = 0.05
critical_value = stats.chi2.ppf(1 - alpha, dof)
print(f"\nCritical value (alpha = {alpha}, df = {dof}) is: {critical_value:.2f}")
if chi2 > critical_value:
    print(f"Since {chi2:.2f} > {critical_value:.2f}, Reject independence -> profession and happiness ARE related")
else:
    print(f"Since {chi2:.2f} <= {critical_value:.2f}, Cannot reject")

# Decide using p-value
if p_value < alpha:
    print(f"p < {alpha}: REJECT independence -> profession and happiness ARE related")
else:
    print(f"p >= {alpha}: cannot reject -> no evidence they are related")


Critical value (alpha = 0.05, df = 7) is: 14.07
Since 936.14 > 14.07, Reject independence -> profession and happiness ARE related
p < 0.05: REJECT independence -> profession and happiness ARE related


Put your analysis in the empty `markdown` cell below.

Ran the chi-square test on profession vs happiness and got chi2 = 936.14, df = 7, p ≈ 7.5e-198. Since p is way under 0.05, we reject independence — profession and happiness are related in this data. The big chi-square number basically means the actual yes/no answers are way off from what you'd expect if profession didn't matter at all, so it's showing a strong link, not "good" or "bad" by itself. And yeah the test is definitely significant, but that doesn't mean profession causes happiness — just that they're connected somehow.